src/raw/List.fs

```diff
@@ -0,0 +1,237 @@
+    let listRowsAsync (database: string) (table: string) (queryParameters: DatabaseRowQuery seq) =
+        listRowsCore database table queryParameters fetch finishEarly
+
+[<Extension>]
+type GetdatabasesByIdsClientExtensions =
+    /// <summary>
+    /// Lists all databases in the same project.
+    /// A maximum of 1000 database names may be listed per request and all
+    /// of them must be unique.
+    /// </summary>
+    /// <param name="queryParameters">Limit and nextCursor</param>
+    /// <returns>databases in project.</returns>
+    [<Extension>]
+    static member ListDatabasesAsync (this: ClientExtension, queryParameters: DatabaseQuery seq, [<Optional>] token: CancellationToken) : Task<_ seq> =
+        task {
+            let ctx = this.Ctx |> Context.setCancellationToken token
+            let! result = Items.listDatabasesAsync queryParameters this.Ctx
+            match result with
+            | Ok ctx ->
+                let databases = ctx.Response
+                return databases |> Seq.map (fun database -> database.ToDatabaseEntity ())
+            | Error error ->
+                return raise (error.ToException ())
+        }
+
+    /// <summary>
+    /// Lists all databases in the same project.
+    /// A maximum of 1000 database names may be listed per request and all
+    /// of them must be unique.
+    /// </summary>
+    /// <returns>databases in project.</returns>
+    [<Extension>]
+    static member ListDatabasesAsync (this: ClientExtension, [<Optional>] token: CancellationToken) : Task<_ seq> =
+        this.ListDatabasesAsync(Seq.empty, token)
+
+    /// <summary>
+    /// Lists all tables in the same project for a given database.
+    /// A maximum of 1000 tables may be listed per request and all
+    /// of them must be unique.
+    /// </summary>
+    /// <param name="queryParameters">Limit and nextCursor</param>
+    /// <returns>tables in project.</returns>
+    [<Extension>]
+    static member ListTablesAsync (this: ClientExtension, database: string, queryParameters: DatabaseQuery seq, [<Optional>] token: CancellationToken) : Task<_ seq> =
+        task {
+            let ctx = this.Ctx |> Context.setCancellationToken token
+            let! result = Items.listTablesAsync database queryParameters this.Ctx
+            match result with
+            | Ok ctx ->
+                let tables = ctx.Response
+                return tables |> Seq.map (fun table -> table.ToTableEntity ())
+            | Error error ->
+                return raise (error.ToException ())
+        }
+
+    /// <summary>
+    /// Lists all tables in the same project for a given database.
+    /// A maximum of 1000 tables may be listed per request and all
+    /// of them must be unique.
+    /// </summary>
+    /// <returns>tables in project.</returns>
+    [<Extension>]
+    static member ListTablesAsync (this: ClientExtension,  database: string, [<Optional>] token: CancellationToken) : Task<_ seq> =
```

Comment:
Instead of having optional values we should overload the method with another version that does not take the cancellation token (and calls the first with CancellationToken.None), or try to set `= default(CancellationToken)`.  Not sure what the default value becomes when not setting it.

Reply:
It might be fine as is, but I'm not sure what the value of the cancellation token is when the code is like this.